compiler/rustc_typeck/src/check/method/mod.rs

```diff
@@ -534,7 +534,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         } else {
             traits::ObligationCause::misc(span, self.body_id)
         };
-        obligations.extend(traits::predicates_for_generics(cause.clone(), self.param_env, bounds));
+        let predicates_cause = cause.clone();
+        obligations.extend(traits::predicates_for_generics(
+            move |_, _| predicates_cause.clone(),
```

Comment:
This clone looks unneeded?

Reply:
It's needed, for example, when you want to write:

```
let a = ...;

let b = f(move || a.clone());

let c = f2(a);
```

Which actually needs to be written:

```
let a = ...;

let a_clone = a.clone();
let b = f(move || a.clone());

let c = f2(a);
```

Reply:
Not?
```rust
let a = ...;

let a_clone = a.clone();
let b = f(move || a_clone);

let c = f2(a);
```
Life is hard.

Reply:
Oh, that's exactly what was changes in `predicates_for_generics` signature, thanks.